ui/lib/password-expiration.ts

```diff
@@ -0,0 +1,23 @@
+import { ethers } from "ethers";
+import { useMemo } from "react";
+import { useAccount } from "./use-wagmi";
+import { useVeNationLock } from "./ve-token";
+
+const threshold = ethers.BigNumber.from(10).pow(18).mul(3).div(2);
+const maxLockPeriod = ethers.BigNumber.from(31556926 * 4);
+
+export function usePassportExpirationDate(): Date | undefined {
+  const { data: account } = useAccount()
+  const { data: veNationLock } =
+    useVeNationLock(account?.address)
+    
+  return useMemo(() => {
+      // threshold <= (lock amount) * (lock end - expiration) / (max lock period)
```

Comment:
What are your best practices for stuff like this? We want people in the future to easily understand why expiration is calculated the way it is